### PR TITLE
#3030 - hide user details when getting apps

### LINKED ISSUE #3030: Hide user details from get applications API
### Version Information
all versions

### What is the expected behaviour?
user's sensitive data should not be available in get apps API `[GET] /api/apps` API is called to list app in home page

### What is the current behaviour?
user's sensitive data is visible

### Screenshots or Screencast
Screenshot 2022-05-10 at 11 15 43 PM

### Any possible solutions?
Empty the fields from back end
- email
- invitation_token
- forgot_password_token
- default_organization_id
- role
- created_at
- updated_at

### If the bug is confirmed, would you be willing to submit a PR?
No

### PR DESCRIPTION
**Pull request by mslourens**
Remove sensitive user details from the apps api response.

Fixes #3030

### CODE REVIEW

File: server/src/controllers/apps.controller.ts [hunk 1]
```diff
@@ -197,8 +199,8 @@ export class AppsController {
     } else {
       apps = await this.appsService.all(user, page, searchKey);
     }
-    //remove password from user info
-    apps.forEach((app) => (app.user.password = undefined));
+    //remove user sensitive data
+    apps.forEach((app) => this.userService.removeSensitiveData(app.user));
```
Comment (gsmithun4) [inline server/src/controllers/apps.controller.ts:203]: Shall we try changing the query?
```
const viewableAppsQb = createQueryBuilder(App, 'apps')
      .select(['app_group_permissions', 'user.first_name', 'user.last_name'])
      .innerJoin('apps.groupPermissions', 'group_permissions')
      .innerJoin('apps.appGroupPermissions', 'app_group_permissions')
      .innerJoin('apps.user', 'user')
      .innerJoin(
        UserGroupPermission,
        'user_group_permissions',
        'app_group_permissions.group_permission_id = user_group_permissions.group_permission_id'
      )
      .where('apps.organization_id = :organizationId', { organizationId: user.organizationId })
      .andWhere(
        new Brackets((qb) => {
          qb.where('user_group_permissions.user_id = :userId', {
            userId: user.id,
          })
            .andWhere('app_group_permissions.read = :value', { value: true })
            .orWhere('apps.is_public = :value OR apps.user_id = :userId', {
              value: true,
              userId: user.id,
            });
        })
      );
    if (searchKey) {
      viewableAppsQb.andWhere('LOWER(apps.name) like :searchKey', {
        searchKey: `%${searchKey && searchKey.toLowerCase()}%`,
      });
    }
```
[Reference](https://github.com/typeorm/typeorm/issues/876)

**Review by gsmithun4 — CHANGES_REQUESTED** [review]

**Comment by withshubh** [comment]
Hey @mslourens 👋 
Please update the PR with the requested changes so that we can merge this fix?

**Comment by withshubh** [comment]
@mslourens 👋 I hope you're fine. Please update the PR with the requested changes?

**Comment by mslourens** [comment]
> @mslourens 👋 I hope you're fine. Please update the PR with the requested changes?

Hi @withshubh, I am fine, but currently a bit busy with my work. Next to that my knowledge of TypeORM and Knex is limited, so if somebody else could update the query, that would be much appreciated.

**Comment by gsmithun4** [comment]
duplicate https://github.com/ToolJet/ToolJet/pull/3722

**Pull request closed without merging**